src/cmds/categories/create.js

```diff
@@ -0,0 +1,121 @@
+  constructor() {
+    this.command = 'categories:create';
+    this.usage = 'categories:create <title> [options]';
+    this.description = 'Create a category with the specified title and guide in your ReadMe project';
+    this.category = 'categories';
+    this.position = 2;
+
+    this.hiddenargs = ['title'];
+    this.args = [
+      {
+        name: 'key',
+        type: String,
+        description: 'Project API key',
+      },
+      {
+        name: 'version',
+        type: String,
+        description: 'Project version',
+      },
+      {
+        name: 'title',
+        type: String,
+        defaultOption: true,
+      },
+      {
+        name: 'categoryType',
+        type: String,
+        description: 'Category type, must be guide or reference',
+      },
+    ];
+  }
+
+  async run(opts) {
+    const { categoryType, title, key, version } = opts;
+    debug(`command: ${this.command}`);
+    debug(`opts: ${JSON.stringify(opts)}`);
+
+    if (!key) {
+      return Promise.reject(new Error('No project API key provided. Please use `--key`.'));
+    }
+
+    if (!title) {
+      return Promise.reject(new Error(`No title provided. Usage \`${config.get('cli')} ${this.usage}\`.`));
+    }
+
+    if (categoryType !== 'guide' && categoryType !== 'reference') {
+      return Promise.reject(new Error('`categoryType` must be guide or reference.'));
+    }
+
+    const selectedVersion = await getProjectVersion(version, key, false);
+
+    debug(`selectedVersion: ${selectedVersion}`);
+
+    const slug = changeCase.paramCase(title);
+
+    function getNumberOfPages() {
+      return fetch(`${config.get('host')}/api/v1/categories?perPage=20&page=1`, {
+        method: 'get',
+        headers: cleanHeaders(key, {
+          'x-readme-version': selectedVersion,
+          Accept: 'application/json',
+        }),
+      }).then(res => {
+        return Math.ceil(res.headers.get('x-total-count') / 20);
+      });
+    }
+
+    const allCategories = [].concat(
+      ...(await Promise.all(
+        Array.from({ length: await getNumberOfPages() }, (_, i) => i + 1).map(async page => {
+          return fetch(`${config.get('host')}/api/v1/categories?perPage=20&page=${page}`, {
+            method: 'get',
+            headers: cleanHeaders(key, {
+              'x-readme-version': selectedVersion,
+              Accept: 'application/json',
+            }),
+          }).then(res => handleRes(res));
+        })
+      ))
+    );
+
+    async function matchCategory() {
+      return allCategories.filter(category => {
+        return category.slug.match(`${slug}(-\\d)?`) && category.type === categoryType;
+      });
+    }
+
+    async function createCategory() {
+      const matchedCategory = await matchCategory();
+      if (matchedCategory.length > 0) {
+        return `The '${matchedCategory[0].slug}' category with a type of '${matchedCategory[0].type}' already exists with an id of '${matchedCategory[0].id}'. A new category was not created`;
```

Comment:
I'm still going back and forth on if this slug uniqueness check is necessary. The category slug isn't surfaced in the docs, it's more of a behind-the-scenes identifier.

Let's say a project contains the category slug `example`. If a user attempts to create the category "Example", the category will be successfully created with a slug of `example-1`. I think as long as we report back what the slug is in the success response, this behavior should be fine—right?

Reply:
Hey @kanadgupta, 

Our development team is publishing documentation to ReadMe via API and are leveraging the `docs` command to publish our guides. Our engineers need to easily create a new or retrieve the existing unique category id for their feature to add to their markdown file's grey-matter. The category title maps to the feature name in our product taxonomy, and we only want one guide category and one reference category per feature. I chose the slug to identify duplicates as it had a more consistent format than title, but suppose the match could be done on the title as well. Curious to hear your thoughts on this.